automation/taskcluster/decision_task.py

```diff
@@ -137,6 +137,43 @@ def release(channel, is_staging, version_name):
     return (build_tasks, signing_tasks, push_tasks)
 
 
+def nightly_to_production_app(is_staging, version_name):
```

Comment:
Nit: This function is really similar to `release()`. I wonder if we shouldn't just expand `release()` to expose  `override_google_play_track`. If I understand correctly, that would allow to let the nightly APK to be shipped on the production track. What do you think?

Reply:
This function looks similar to `release()`, I agree, but I kept it as a separate function for a few reasons:

1. `release()` captures the regular release process (prod built, prod channel, deploy to prod app). Meanwhile, the legacy "nightly app to production app" logic is here for legacy reasons. So, these two functions serve mildly different purposes, and will most-likely change at different rates
2. Having it as a separate function calls to attention the exceptional/different nature of this nightly release.
3. We could try to represent "regular release process + different release process" in a single function, but that would make that one function less coherent

For these reasons, I feel that having this as a separate function is improving maintainability

Reply:
Discussed offline: The current flow is optimized for people trying to understand how a nightly (with the prod key) is made, which seems the most probable one. Okay to keep it as is.